custom/enikshay/management/commands/investigations_reconciliation.py

```diff
@@ -0,0 +1,258 @@
+            bulk_update_cases(DOMAIN, updates, self.__module__)
+
+    def get_coalesced_value_for_case_property(self, case_property, all_values):
+        #ToDo: write logic for this
+        return all_values[0]
+
+    def episode_case_needs_reconciliation(self, episode_case):
+        # iterate investigation cases to check if any investigation type has more than 1
+        # investigation case added for it
+        investigation_cases = get_investigation_cases_from_episode(episode_case)
+        investigation_cases_by_interval = defaultdict(list)
+        for investigation_case in investigation_cases:
+            investigation_case_interval = investigation_case.get_case_property("investigation_interval")
+            if investigation_case_interval not in self.investigation_interval_values:
+                self.investigation_interval_values.append(investigation_case_interval)
+            investigation_cases_by_interval[investigation_case_interval].append(
+                investigation_case
+            )
+        investigation_cases_to_reconcile = {}
+        for interval_type, investigation_cases in investigation_cases_by_interval.items():
+            if len(investigation_cases) > 1:
+                investigation_cases_to_reconcile[interval_type] = investigation_cases
+
+        return investigation_cases_to_reconcile
+
+    @staticmethod
+    def public_app_case(person_case):
+        if person_case.get_case_property(ENROLLED_IN_PRIVATE) == 'true':
+            return False
+        return True
+
+    @staticmethod
+    def _get_open_person_case_ids_to_process():
+        from corehq.sql_db.util import get_db_aliases_for_partitioned_query
+        dbs = get_db_aliases_for_partitioned_query()
+        for db in dbs:
+            case_ids = (
+                CommCareCaseSQL.objects
+                .using(db)
+                .filter(domain=DOMAIN, type="person", closed=False)
+                .values_list('case_id', flat=True)
+            )
+            num_case_ids = len(case_ids)
+            print("processing %d docs from db %s" % (num_case_ids, db))
+            for i, case_id in enumerate(case_ids):
+                yield case_id
+                if i % 1000 == 0:
+                    print("processed %d / %d docs from db %s" % (i, num_case_ids, db))
+
+
+def get_all_occurrence_cases_from_person(person_case_id):
+    case_accessor = CaseAccessors(DOMAIN)
+    all_cases = case_accessor.get_reverse_indexed_cases([person_case_id])
+    return [case for case in all_cases if case.type == CASE_TYPE_OCCURRENCE]
+
+
+def get_open_confirmed_drtb_episode_cases(person_case_id):
+    occurrence_cases = get_all_occurrence_cases_from_person(
+        person_case_id
+    )
+    open_confirmed_drtb_episode_cases = []
+    for occurrence_case in occurrence_cases:
+        case_accessor = CaseAccessors(DOMAIN)
+        all_cases = case_accessor.get_reverse_indexed_cases([occurrence_case.case_id])
```

Comment:
I think this could just accept a list of all the occurrence case ids, rather than doing this in a loop and extending the list.

Reply:
Oh right. Updated!